cocos/core/assets/effect-asset.ts

```diff
@@ -294,6 +295,14 @@ export class EffectAsset extends Asset {
     public validate () {
         return this.techniques.length > 0 && this.shaders.length > 0;
     }
+
+    private modifyDescriptorSetLayout () {
+        //set max joints after device initialize.
+        const root = legacyCC.director.root as Root;
+        let maxJoints = Math.floor((root.device.capabilities.maxVertexUniformVectors - 38) / 3);
+        maxJoints = maxJoints < 256 ? maxJoints : 256;
+        localDescriptorSetLayout_ResizeMaxJoints(maxJoints);
```

Comment:
This is too hack, in general EffectAsset, we shouldn't care about the joint layout or max joints availability, we should move the check to other places